src/core/meshcop/joiner_router.cpp

```diff
@@ -315,18 +315,19 @@ Coap::Message *JoinerRouter::PrepareJoinerEntrustMessage(void)
     Error          error;
     Coap::Message *message = nullptr;
     Dataset        dataset;
-
+    NetworkKey     networkKey;
     NetworkNameTlv networkName;
     const Tlv *    tlv;
 
     VerifyOrExit((message = NewMeshCoPMessage(Get<Tmf::Agent>())) != nullptr, error = kErrorNoBufs);
+    IgnoreError(Get<KeyManager>().GetNetworkKey().CopyKey(networkKey.m8, OT_NETWORK_KEY_SIZE));
 
     message->InitAsConfirmablePost();
     SuccessOrExit(error = message->AppendUriPathOptions(UriPath::kJoinerEntrust));
     SuccessOrExit(error = message->SetPayloadMarker());
     message->SetSubType(Message::kSubTypeJoinerEntrust);
 
-    SuccessOrExit(error = Tlv::Append<NetworkKeyTlv>(*message, Get<KeyManager>().GetNetworkKey()));
+    SuccessOrExit(error = Tlv::Append<NetworkKeyTlv>(*message, networkKey));
```

Comment:
I think the change to `NetworkKey`/`Pskc` definitions (i.e., adding additional member variables to it compared to `otNetworkKey`/`otPskc`) will create other issues/complications as well.

For example with this change, the definitions of `MeshCoP:: NetworkKeyTlv` or `PskcTlv` are not longer valid since they are using `NetworkKey`/`Pskc` as the TLV value (which are now larger than literal key and contain additional data fields).

For example, I think this line `Tlv::Append<NetworkKeyTlv>(*message, networkKey)` here won't be correct and will append the entire `NetworkKey` object (which now has additional fields) as part of the TLV's value.

Reply:
Ok I will separate out the NetwokKey and Pskc from the key refs as suggested.